cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskPlanner.java

```diff
@@ -382,11 +382,17 @@ private void maybeAddLeaderChangeTasks(Collection<ExecutionProposal> proposals,
               && !partitionsInvolved.contains(tp)) {
             partitionsInvolved.add(tp);
             executableReplicaMovements.add(task);
-            // Record the two brokers as involved in this round and stop involving them again in this round.
+            // Record the brokers as involved in this round and stop involving them again in this round.
             brokerInvolved.add(sourceBroker);
             brokerInvolved.addAll(destinationBrokers);
+            // The first task of each involved broker might have changed.
+            // Let's remove the brokers before the tasks change, then add them again later by comparing their new first tasks.
+            _interPartMoveBrokerIds.remove(sourceBroker);
```

Comment:
1. Given _interPartMoveBrokerIds is a SortedSet<Integer>, I don't understand why we need to remove and add back the source and destinationBrokers. Is that not an no-op in the end?


2. It seems the _interPartMoveBrokerIds variable can be turned into a local variable inside this method from a field in this class. It's being set to an empty set inside the maybeAddInterBrokerReplicaMovementTasks method, but all we need to save in that method is the comparator.